tools/ci_build/github/azure-pipelines/templates/telemetry-steps.yml

```
@@ -3,7 +3,8 @@
 
 steps:
   - powershell: |
-      if ($env:TELEMETRYGUID) {
+      $telemetry_guid = ${{ variables['TELEMETRYGUID'] }}
+      if ( -not [string]::IsNullOrEmpty( $telemetry_guid ) ){
         $length = $env:TELEMETRYGUID.length
         $fileContent = "#define TraceLoggingOptionMicrosoftTelemetry() \
           TraceLoggingOptionGroup("+$env:TELEMETRYGUID.substring(1, $length-2)+")"
```

do we need to replace other uses of $env:TELEMETRYGUID?
```suggestion
        $length = $telemetry_guid.length
        $fileContent = "#define TraceLoggingOptionMicrosoftTelemetry() \
          TraceLoggingOptionGroup("+$telemetry_guid.substring(1, $length-2)+")"
```